### PR TITLE
More updates of Readme

### PR DESCRIPTION
**Pull request by MaximProshin**
### Changes

This PR updates the Readme file with the Contents, more examples, corrections for PTQ, slight re-structure, etc.
More changes will come in further PRs for other NNCF documents. I decided to split it to reduce the number of potential comments and review time. 

### Reason for changes

Docs improvements

### Related tickets

na

### Tests

na

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -303,12 +334,13 @@ See [third_party_integration](./third_party_integration) for examples of code mo
 - Ubuntu\* 18.04 or later (64-bit)
 - Python\* 3.7 or later
 - Supported frameworks:
-  - PyTorch\* 1.12.1
+  - PyTorch\* >=1.9.1, <1.14
   - TensorFlow\* >=2.4.0, <=2.11.1
+  - onnx~=1.13.1, onnxruntime~=1.14.1
```
Comment (alexsu52) [inline README.md:339]:
```suggestion
  - ONNX\* ~=1.13.1
  - OpenVINO\* >=2022.3.0
```

File: README.md [hunk 2]
```diff
@@ -494,14 +526,11 @@ to find instruction and links to exact configuration files and final checkpoints
 ```
 
 ## Useful links
-- [Documentation](./docs)
+- [NNCF Documentation](./docs)
```
Comment (alexsu52) [inline README.md:529]: Are you sure?

File: README.md [hunk 3]
```diff
@@ -494,14 +526,11 @@ to find instruction and links to exact configuration files and final checkpoints
...
 - Example scripts (model objects available through links in respective README.md files):
     - [PyTorch](./examples/torch)
     - [TensorFlow](./examples/tensorflow)
-- [FAQ](./docs/FAQ.md)
+- [NNCF FAQ](./docs/FAQ.md)
```
Comment (alexsu52) [inline README.md:533]: Are you sure?

File: README.md [hunk 4]
```diff
@@ -9,14 +9,28 @@
  
 </div>
 
-_For the installation instructions, [click here](#installation)._
+## Contents:
```
[the remark this answers is not in the document]
Comment (MaximProshin) [reply README.md:12]: https://github.com/openvinotoolkit/openvino
It was also specifically requested several times as we have a lot of contents in the Readme.

File: README.md [hunk 5]
```diff
@@ -27,39 +41,40 @@ learning frameworks.
...
+- Seamless combination of pruning, sparsity and quantization algorithms. Please refer to [optimum-intel](https://github.com/huggingface/optimum-intel/tree/main/examples/openvino) for examples of 
 joint (movement) pruning, quantization and distillation (JPQD), end-to-end from NNCF optimization to compressed OpenVINO IR.
-- Exporting PyTorch compressed models to ONNX\* checkpoints and TensorFlow compressed models to SavedModel or Frozen Graph format, ready to use with [OpenVINO&trade; toolkit](https://docs.openvino.ai/latest/home.html).
+- Exporting PyTorch compressed models to ONNX\* checkpoints and TensorFlow compressed models to SavedModel or Frozen Graph format, ready to use with [OpenVINO&trade; toolkit](https://docs.openvino.ai).
 - Support for [Accuracy-Aware model training](./docs/Usage.md#accuracy-aware-model-training) pipelines via the [Adaptive Compression Level Training](./docs/accuracy_aware_model_training/AdaptiveCompressionLevelTraining.md) and [Early Exit Training](./docs/accuracy_aware_model_training/EarlyExitTraining.md).
 
+## Documentation
```
[the remark this answers is not in the document]
Comment (MaximProshin) [reply README.md:70]: I don't follow your request. If you provide an example, it would be clearer.

File: README.md [hunk 6]
```diff
@@ -9,14 +9,28 @@
...
+
+## What is Neural Network Compression Framework?
+
+Neural Network Compression Framework (NNCF) provides a suite of post-training and training-time algorithms for neural networks inference optimization in [OpenVINO&trade;](https://docs.openvino.ai) with minimal accuracy drop.
 
 NNCF is designed to work with models from [PyTorch](https://pytorch.org/), [TensorFlow](https://www.tensorflow.org/), [ONNX](https://onnx.ai/) and [OpenVINO&trade;](https://docs.openvino.ai/latest/home.html).
 
-NNCF provides samples that demonstrate the usage of compression algorithms for three different use cases on public PyTorch and 
-TensorFlow models and datasets: Image Classification, Object Detection and Semantic Segmentation. 
+NNCF provides [examples](#usage) and [samples](#Model-Compression-Samples) that demonstrate the usage of compression algorithms for different use cases and models.
```
[the remark this answers is not in the document]
Comment (MaximProshin) [reply README.md]: it's because we have different sections for them. I've moved it back to avoid any confusion.

File: README.md (hunk 3 above)
Comment (MaximProshin) [reply README.md:534]: For me it looks better as we also refer to openvino, source frameworks, etc a lot and it's unclear which documentation or FAQ we refer to. If you strongly object, I can move it back, no problem. Let's see what others say.

File: README.md [hunk 7]
```diff
@@ -303,12 +334,14 @@ See [third_party_integration](./third_party_integration) for examples of code mo
 - Ubuntu\* 18.04 or later (64-bit)
 - Python\* 3.7 or later
 - Supported frameworks:
-  - PyTorch\* 1.12.1
+  - PyTorch\* >=1.9.1, <1.14
   - TensorFlow\* >=2.4.0, <=2.11.1
```
Comment (KodiaqQ) [inline README.md:338]: From setup.py - `tensorflow~=2.11.1`

Comment (MaximProshin) [reply README.md:338]: good catch! @alexsu52 , we should sort it out

Comment (MaximProshin) [reply README.md:338]: @KodiaqQ , I think in documents it's written correctly. setup.py should probably be updated but not in this PR

File: README.md (hunk 3 above)
Comment (vshampor) [reply README.md:534]: It's our repo, so plain "Documentation" should refer to our docs. For links to docs of other products, prefix accordingly, i.e. "OpenVINO docs" etx.

File: README.md [hunk 8]
```diff
@@ -9,14 +9,28 @@
  
 </div>
 
-_For the installation instructions, [click here](#installation)._
```
Comment (vshampor) [inline README.md:12]: IMO, the first thing that the user looks for when opening a repo is how to install it. This line serves that very purpose.

File: README.md [hunk 9]
```diff
@@ -27,39 +41,40 @@ learning frameworks.
 ## Key Features
 ### Post-Training Compression Algorithms
 
-| Compression algorithm                                                       |PyTorch|TensorFlow|   ONNX   |       OpenVINO     |
+| Compression algorithm                                                       |OpenVINO|PyTorch|   TensorFlow   |     ONNX       |
 |:----------------------------------------------------------------------------| :---: | :---: |:--------:|:------------------:|
-| [Quantization](./docs/compression_algorithms/post_training/Quantization.md) | Supported | Supported |Supported| Preview |
-
-_Preview means that this is a work in progress and NNCF does not guarantee the full functional support._
+| [Post-Trainig Quantization](./docs/compression_algorithms/post_training/Quantization.md) | Supported | Supported |Supported| Supported |
```
Comment (vshampor) [inline README.md:46]: `Post-Training`

File: README.md [hunk 10]
```diff
@@ -9,14 +9,28 @@
...
+- [Documentation](#documentation)
+- [Usage examples](#usage)
+- [Model compression tutorials and samples](#Model-compression-tutorials-and-samples)
+- [Third-party repository integration](#Third-party-repository-integration)
+- [System requirements](#System-requirements)
+- [Installation guide](#Installation-guide)
+- [Contributing guide](#Contributing-Guide)
+- [NNCF Compressed Model Zoo](#NNCF-Compressed-Model-Zoo)
+- [Citing](#Citing)
+- [Useful links](#Useful-links)
```
Comment (vshampor) [inline README.md:25]: These links aren't working, you're supposed to write references to header anchors in lowercase only.

**Review by vshampor — CHANGES_REQUESTED** [review]
More or less fine overall, some comments are to be addressed, though.

File: README.md (hunk 8 above)
Comment (MaximProshin) [reply README.md:12]: I know, but now they can do it quickly from the Contents. Do you insist to have a dedicated line for it in the first place?

Comment (vshampor) [reply README.md:12]: The link in the contents is visually buried between the other links and does not immediately stand out, unlike the dedicated line.

File: README.md (hunk 7 above)
Comment (MaximProshin) [reply README.md:338]: agreed to fix it in develop by a separate PR

File: README.md (hunk 3 above)
Comment (MaximProshin) [reply README.md]: Ok, moved it back. For me it's not crucial

File: README.md (hunk 2 above)
Comment (MaximProshin) [reply README.md]: Ok, moved it back. For me it's not crucial

File: README.md (hunk 8 above)
Comment (MaximProshin) [reply README.md:12]: Ok, moved it back.

**Review by vshampor — APPROVED** [review]

**Comment by MaximProshin** [comment]
@vshampor , please check it again. I changes the way how Contents is presented and now it's in the header where "Installation" is pretty obvious. So I think it's better to remove the line _For the installation instructions, [click here]_
